src/services/Store/store/membership.slice.ts

```diff
@@ -87,14 +93,33 @@ export function* fetchMembershipsSaga() {
 }
 
 export function* fetchMembershipsWorker({ payload }: PayloadAction<IAccount[] | undefined>) {
-  const accounts: StoreAccount[] = yield select(getWalletAccountsOnDefaultNetwork);
-  const network: Network = yield select(selectDefaultNetwork);
+  const accounts: StoreAccount[] = yield select(getAccounts);
+  const membershipNetworkIds = [
+    ...new Set(Object.values(MEMBERSHIP_CONFIG).map(({ networkId }) => networkId))
+  ];
+  const membershipNetworkAccounts = accounts.filter(({ networkId }) =>
+    membershipNetworkIds.includes(networkId)
+  );
+  const ethereumNetwork: Network = yield select(selectDefaultNetwork);
+  const xdaiNetwork: Network = yield select(selectNetwork(XDAI_NETWORK));
+  const xdaiAccounts = (payload || membershipNetworkAccounts).filter((a) =>
+    isAccountInNetwork(a, XDAI_NETWORK)
+  );
 
-  const relevantAccounts = (payload ?? accounts).map((a) => a.address);
+  const ethereumAccounts = (payload || membershipNetworkAccounts).filter(isEthereumAccount);
 
   try {
-    const memberships = yield call(MembershipApi.getMemberships, relevantAccounts, network);
-    yield put(slice.actions.setMemberships(memberships));
+    const ethereumMemberships = yield call(
+      MembershipApi.getMemberships,
+      ethereumAccounts.map(({ address }) => address as TAddress),
+      ethereumNetwork
+    );
+    const xdaiMemberships = yield call(
+      MembershipApi.getMemberships,
+      xdaiAccounts.map(({ address }) => address as TAddress),
```

Comment:
Can we add some XDAI accounts to the test to make sure that everything works?

Reply:
I don't think that's really necessary as we're mocking the `getMemberships` call/results anyways.

Reply:
Yes, but this means that the xDAI code could not work and we wouldn't spot it. So I would prefer if it ran 😄